### PR TITLE
[PDI-331] Add javascript & typescript webhook verification example

### PR DESCRIPTION
**Pull request by tl-alex-butler**
Add a js & typescript webhook verification binary using the `jws`, `jwks-rsa` & `js-base64` libs.

I've moved the request signing bin into `./sign` directory, with the new bin in `./verify` for both javascript & typescript.

Also update javascript/typescript dependencies.

### CODE REVIEW

File: javascript/verify/src/main.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    // Construct a full jws using the `body` & detached jws
+    const signatureParts = tlSignature.split('.');
+    const jws = `${signatureParts[0]}.${Base64.encode(body, true)}.${signatureParts[2]}`;
+
+    const { header } = Jws.decode(jws);
+    const publicKey = await fetchJwksPublicKey(header.jku, header.kid);
+
+    // Verify using the public key
+    if (!Jws.verify(jws, header.alg, publicKey))
+        throw new Error("Invalid signature");
```
Comment (stefandanaita) [inline javascript/verify/src/main.js:45]: { } man 😂

Comment (tl-alex-butler) [reply javascript/verify/src/main.js:45]: the js parser says I'm cool to do this :smile:

Comment (stefandanaita) [reply javascript/verify/src/main.js:45]: image

File: typescript/sign/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "description": "",
+  "main": "src/main.ts",
+  "scripts": {
+    "signing": "TS_NODE_PROJECT='./tsconfig.json' ts-node ./src/main.ts"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/TrueLayer/paydirect-signing-examples.git"
+  },
+  "author": "David Hayden <david.hayden@truelayer.com>",
```
Comment (stefandanaita) [inline typescript/sign/package.json:13]: Suspicious 🕵🏻‍♂️

Comment (tl-alex-butler) [reply typescript/sign/package.json:13]: I just moved the sign example, it already existed

File: typescript/verify/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    // Construct a full jws using the `body` & detached jws
+    const signatureParts = tlSignature.split('.');
+    const jws = `${signatureParts[0]}.${Base64.encode(body, true)}.${signatureParts[2]}`;
+
+    const { header } = Jws.decode(jws);
+    const publicKey = await fetchJwksPublicKey(header.jku, header.kid);
+
+    // Verify using the public key
+    if (!Jws.verify(jws, header.alg, publicKey))
+        throw new Error("Invalid signature");
```
Comment (stefandanaita) [inline typescript/verify/src/main.ts:52]: AAAAAAAAAAAAAAAAAAAAAAAA

**Review by stefandanaita — DISMISSED** [review]
Looks good 👌🏻

File: javascript/verify/src/main.js (hunk 1 above)
Comment (tl-alex-butler) [reply javascript/verify/src/main.js:45]: I've added braces!

**Review by stefandanaita — APPROVED** [review]
Looks better! 😄

**Pull request merged**